src/scf/self_consistent_field.jl

```diff
@@ -62,6 +66,27 @@ function scf_convergence_density_difference(tolerance)
     info -> norm(info.ρout.fourier - info.ρin.fourier) < tolerance
 end
 
+"""
+Determine the tolerance used for the next diagonalisation. This function takes
+``|ρnext - ρin|`` and multiplies it with `ratio_ρdiff` to get the next `diagtol`,
+ensuring additionally that the returned value is between `diagtol_min` and `diagtol_max`.
+"""
+function scf_determine_diagtol(ratio_ρdiff=0.2, diagtol_min=nothing, diagtol_max=0.1)
+    function determine_diagtol(neval, norm_ρnext_ρin)
```

Comment:
shouldn't it just take `info` like the callbacks?

Reply:
yes. Turned out to be simpler than I thought to do that.